apps/desktop-wallet/src/features/tokenPrices/hooks.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash, Asset, calculateAmountWorth, client, ONE_MINUTE_MS, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { ALPH } from '@alephium/token-list'
+import { explorer } from '@alephium/web3'
+import { useQueries } from '@tanstack/react-query'
+import { chunk, orderBy } from 'lodash'
+import { useMemo } from 'react'
+
+import { useAppSelector } from '@/hooks/redux'
+import {
+  makeSelectAddressesKnownFungibleTokens,
+  makeSelectAddressesListedFungibleTokenSymbols
+} from '@/storage/addresses/addressesSelectors'
+
+export const useAddressTokensPrices = () => {
+  const currency = useAppSelector((s) => s.settings.fiatCurrency).toLowerCase()
+  const addressTokensSymbols = useAppSelector(useMemo(makeSelectAddressesListedFungibleTokenSymbols, [])) // TODO: To be replaced when tokens are fetched with Tanstack
+  const addressTokensSymbolsWithPrice = addressTokensSymbols.filter((symbol) => symbol in explorer.TokensWithPrice)
+
+  const { data, isPending } = useQueries({
+    queries: chunk(addressTokensSymbolsWithPrice, TOKENS_QUERY_LIMIT).map((symbols) => ({
+      queryKey: ['tokenPrices', symbols, { currency }],
+      queryFn: async () =>
+        (await client.explorer.market.postMarketPrices({ currency }, symbols)).map((price, i) => ({
+          price,
+          symbol: symbols[i]
+        })),
+      refetchInterval: ONE_MINUTE_MS
+    })),
+    combine: (results) => ({
+      data: results.flatMap(({ data }) => data).filter((price) => !!price),
+      isPending: results.some(({ isPending }) => isPending)
+    })
+  })
+
+  return { data, isPending }
+}
```

Comment:
Hehe, I was having a hard time deciding as well. I guess since this triggers something for you, the best would be to stick with the pattern that we have created: If it handles an array, then use plural, if not, singular. I'll update.